evaluator/evaluator.go

```diff
@@ -86,7 +105,24 @@ func evalStatements(backend Backend, stmts []ast.Statement) object.Object {
 	return result
 }
 
+// concatenateRows by simply splicing the tuples together.
+// Used when joining tables
+func concatenateRows(row1 object.Row, row2 object.Row) object.Row {
```

Comment:
This probably works, but should use a new struct type that's for rows from the backend